### PR TITLE
Adds a way to programmatically set camera focus (on one node)

### PR DESCRIPTION
**Pull request by neil-morrison44**
## PR Type
```
[x] Feature
```

## What is the current behavior?

Currently there's no way to programmatically set the camera focus, so setting selection via click can do more than setting selection elsewhere.

Issue Number: https://github.com/reaviz/reagraph/issues/27

## What is the new behavior?

Adds a `focusOnNode` method to the `useSelection` hook, allowing focus to be set externally.
I considered having the `setSelection`,  `addSelection` etc methods check `focusOnSelect` within themselves but that was a bigger change than I wanted to make ( though if you want to do that instead feel free to go down that route & close this PR).

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/helpers/useSelection.ts [hunk 1]
```diff
@@ -172,6 +177,10 @@ export const useSelection = ({
     clearSelections(path.map(p => p.id as string));
   };
 
+  const focusOnNode = (item: string) => {
+    ref.current?.centerGraph([item]);
```
Comment (Singh212dab) [inline src/helpers/useSelection.ts:181]: ref.current?.centerGraph(item);

**Review by Singh212dab — CHANGES_REQUESTED** [review]

File: src/helpers/useSelection.ts (hunk 1 above)
Comment (neil-morrison44) [reply src/helpers/useSelection.ts:181]: Gets passed an array, see:
https://github.com/reaviz/reagraph/blob/master/src/helpers/useSelection.ts#L156
```
ref.current?.centerGraph([data.id]);
```

**Review by amcdnl — CHANGES_REQUESTED** [review]
See feedback on the issue

File: src/helpers/useSelection.ts (hunk 1 above)
Comment (amcdnl) [reply src/helpers/useSelection.ts:181]: We could make this smarter so u don't have to pass an array if you think that would be helpful.

Comment (neil-morrison44) [reply src/helpers/useSelection.ts:181]: I think it’s fine - supports the use cases where the graph gets centred on multiple nodes

**Comment by neil-morrison44** [comment]
Going to close this & the issue out - realised that doing 

```ts
graphRef.current?.centerGraph([id])
```
&
```ts
graphRef.current?.centerGraph([])
```

On the consuming side gives me all the programmatic focus control I need

**Pull request closed without merging**